libs/platform/ebpf_handle.h

```diff
@@ -38,7 +38,7 @@ extern "C"
      *  operation.
      */
     _Must_inspect_result_ ebpf_result_t
-    ebpf_handle_create(ebpf_handle_t* handle, struct _ebpf_base_object* object);
+    ebpf_handle_create(_Out_ ebpf_handle_t* handle, _In_ struct _ebpf_base_object* object);
```

Comment:
Shouldn't the ```_In_``` parameter here (and elsewhere in this PR) really be ```_InOut_```?

AFAIK ```_In_``` parameters _almost_ always require ```const``` as well, but in this case the ```object``` parameter _is_ indirectly modified in ```acquire_reference()```, so ```_InOut_``` is more appropriate.

Reply:
Apologies for the typo.  That should be ```_Inout_```

Reply:
[sal-basics](https://learn.microsoft.com/en-us/cpp/code-quality/understanding-sal?view=msvc-170#sal-basics)
![image](https://user-images.githubusercontent.com/20480683/207202737-60da7982-85f3-45ca-9bbb-9e482e4b4886.png)

General rule is:
Const pointer implies _In_ 
Modifiable pointer implies _Inout_
If the pointer is invalid on input, then _Out_

In this case, the call to ebpf_handle_create modified the object (by incrementing its ref-count), so _Inout_ probably makes more sense.

Reply:
Thank you both for noting. I just updated the files.